oarepo_vocabularies/ui/theme/assets/semantic-ui/js/oarepo_vocabularies_ui/utils.js

```diff
@@ -0,0 +1,85 @@
+// TODO: this utility could be moved to oarepo-ui
+export const getInputFromDOM = (elementName) => {
+  const element = document.getElementsByName(elementName);
+  if (element.length > 0 && element[0].hasAttribute("value")) {
+    return JSON.parse(element[0].value);
+  }
+  return null;
+};
+
+// functiona that temporarily resolves getting vocabulary type from url
+export const extractVariablePart = (url) => {
+  const regex = /\/vocabularies\/([^/]+)/;
+  const match = url.match(regex)?.[1];
+  return match || null; // or any default value you prefer
+};
+
+// function that transforms the shape of fields with multiple values to shape accepted by api
+
+export const transformArrayToObject = (arr) => {
+  const result = {};
+
+  arr.forEach((obj) => {
+    const { language, title } = obj;
+    result[language] = title;
+  });
+
+  return result;
+};
+
+// function that creates array suitable for formik arrayfield
+export const translateObjectToArray = (obj) => {
+  const result = [];
+
+  Object.keys(obj).forEach((language) => {
+    const title = obj[language];
+    result.push({ language, title });
+  });
+
+  return result;
+};
+
+//   function to clear all values in an object
+
+export const clearObjectValues = (obj) => {
+  const result = {};
+
+  for (let key in obj) {
+    result[key] = "";
```

Comment:
If I understand well, you mean to check if the result[key] is Object and then return and empty object and if it is array to return empty array. The problem is, in our case, all of the values are actually objects and my goal is to have simple string fields, which means if I apply approach to return empty object of the value is object, my inputs that don't have some actual value will be filled by [object Object], as they are all actually objects 

![image](https://github.com/oarepo/oarepo-vocabularies/assets/34883098/7a7d9a3d-5d21-4644-9573-81fbf3dcd03b)

Reply:
OK I get the point now :) But still this function doesn't seem to work, when I go to e.g.: https://0.0.0.0:5000/vocabularies/languages/ab/edit and click on Reset form, nothing happens, all values stays in the Formik state

Reply:
This functionality was not meant to clear all values from the form, but rather to eliminate fields that are empty strings before submitting, because sending prop fields like for example ICO with value "" would actually create this field in this item's schema. Anyway, I cleaned up the code and this function is not going to be used at all, so I will mark this as resolved :) The reset button is simple formik's reset form that resets the form back to its initialstate.